framework/src/mfem/mesh/MFEMMesh.C

```diff
@@ -93,23 +93,7 @@ MFEMMesh::displace(mfem::GridFunction const & displacement)
 void
 MFEMMesh::buildDummyMooseMesh()
 {
-  auto element = new Quad4;
-  element->set_id() = 1;
-  element->processor_id() = 0;
-
-  getMesh().add_elem(element);
-
-  Point pt1(0.0, 0.0, 0.0);
-  Point pt2(1.0, 0.0, 0.0);
-  Point pt3(1.0, 1.0, 0.0);
-  Point pt4(0.0, 1.0, 0.0);
-
-  element->set_node(0, getMesh().add_point(pt1));
-  element->set_node(1, getMesh().add_point(pt2));
-  element->set_node(2, getMesh().add_point(pt3));
-  element->set_node(3, getMesh().add_point(pt4));
-
-  getMesh().prepare_for_use();
+  MeshTools::Generation::build_point(static_cast<UnstructuredMesh &>(getMesh()));
```

Comment:
I just opened https://github.com/libMesh/libmesh/pull/4203 so we can avoid casts like this in the future